add support for mongodb and mongoose

hezronkimutai opened the pull request:
#### What does this PR do?
This PR adds support for Mongo DB
#### Description of Task to be completed?
- Added a script that will be used to generate user controllers
- Added a script that will be used to generate user model
- Added a script that will be used to generate MongoDB set up
#### How should this be manually tested?
- fetch the latest changes
- checkout to branch `add-MongoDB-support`
- `npm start init`
- Ensure you select mongoDB as your database and mongoose as your orm
- observe the generated app and you will see that MongoDB is set up 
#### Any background context you want to provide?
The impelmentation of this task was based on [this](https://medium.com/@yugagrawal95/mongoose-mongodb-functions-for-crud-application-1f54d74f1b34) post
#### Screenshots (if appropriate)
![Alt Text](https://res.cloudinary.com/hezzie/image/upload/v1581970424/ezgif.com-video-to-gif_2_ifekyx.gif)

its-tayo commented on tasks/controllers/mongodbcontroller.js:
```diff
@@ -0,0 +1,80 @@
+      message: "An error occurred while creating user"
+    });
+  }
+};
```
Hi @hezronkimutai, I think we should wrap all async code inside a function that catches the errors instead of having multiple try-catch blocks. This function could be in a helpers file or something, like so

```
exports.catchErrors = (fn) => {
  return function(req, res, next) {
    return fn(req, res, next).catch(next);
  };
};
```

This way we can wrap all async controllers in the routes file like so:
```
router.route('/').post(
    UserMiddleware,
    catchErrors(UserController.createUser)
);
```

ezkemboi replied:
@its-tayo I totally agree with this suggestion.

hezronkimutai replied:
@its-tayo, this is really nice. I will do that right away.

hezronkimutai replied:
Do you suggest I put that function in the helper or in the middleware folder?

its-tayo replied:
@hezronkimutai I think the helper folder is more appropriate.

hezronkimutai replied:
Okey thanks @its-tayo .

hezronkimutai commented:
Hi @its-tayo  and @ezkemboi , I have implemented the feedback I got. I created a helper function that handles 500 errors, I also modified the routes such that it the error handler is passed as middleware.

ezkemboi commented:
Awesome @hezronkimutai, I will re-check and test this. 
At the moment, consider resolving/rebase the merge conflicts. 
Thanks.
